src/backend/distributed/commands/multi_copy.c

```diff
@@ -956,13 +956,25 @@ OpenCopyTransactions(CopyStmt *copyStatement, ShardConnections *shardConnections
 		ShardPlacement *placement = (ShardPlacement *) lfirst(placementCell);
 		char *nodeName = placement->nodeName;
 		int nodePort = placement->nodePort;
+		WorkerNode *workerNode = FindWorkerNode(nodeName, nodePort);
+		int workerGroupId = 0;
 		char *nodeUser = CurrentUserName();
 		PGconn *connection = ConnectToNode(nodeName, nodePort, nodeUser);
 
 		TransactionConnection *transactionConnection = NULL;
 		StringInfo copyCommand = NULL;
 		PGresult *result = NULL;
 
+		/*
+		 * When a copy is initiated from a worker, the information about the connected
+		 * worker node may not be found if pg_dist_node entries are not synced to this
+		 * node.
```

Comment:
To elaborate further, maybe "In that case we leave the groupId as 0. Fortunately, it is unused since COPY from a worker does not initiate a 2PC."